### PR TITLE
fix: add index.ts and make package.json refer to it in main field

### PR DESCRIPTION
**Pull request by lehnerm**
## Summary

Adds an src/index.ts file exporting the cli's runMigration
function. Fixes eslint/no-unresolved linter errors in 
dependent projects and makes using the cli as a library easier.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -142,7 +142,7 @@ npm install contentful-migration
 ### Usage as a library
 
 ```javascript
-const runMigration = require('contentful-migration/built/bin/cli').runMigration
+const runMigration = require('contentful-migration')
```
Comment (suevalov) [inline README.md:145]: I would avoid default export in favor of 

```
const { runMigration } = require('contentful-migration');
```

so we can make non-breaking changes if we want export something else

Comment (lehnerm) [reply README.md]: :+1:

**Comment by suevalov** [comment]
We need to update `index.d.ts` to reflect a new method being exported.

**Comment by lehnerm** [comment]
When this PR is ready, what is the process to release it @phoebeschmidt? I'm not allowed to merge.

**Comment by phoebeschmidt** [comment]
@lehnerm Ah okay, I will merge now and with properly formed commit messages, semantic release will release it to npm. 

I'll ask my manager about adding other teams so they have rights to merge

**Pull request merged**